Tasks/Common/webdeployment-common/deployusingmsdeploy.ts

```diff
@@ -22,28 +23,41 @@ export async function DeployUsingMSDeploy(webDeployPkg, webAppName, publishingPr
         excludeFilesFromAppDataFlag, takeAppOfflineFlag, virtualApplication, setParametersFile, additionalArguments, isFolderBasedDeployment, useWebDeploy) {
 
     setParametersFile = utility.getSetParamFilePath(setParametersFile);
+    var setParametersFileName = null;
+    var pathVar;
+    if(setParametersFile != null) {
+        setParametersFileName = setParametersFile.slice(setParametersFile.lastIndexOf('\\') + 1, setParametersFile.length);
+    }
     var isParamFilePresentInPackage = isFolderBasedDeployment ? false : await msDeployUtility.containsParamFile(webDeployPkg);
     var msDeployPath = await msDeployUtility.getMSDeployFullPath();
     var msDeployCmdArgs = msDeployUtility.getMSDeployCmdArgs(webDeployPkg, webAppName, publishingProfile, removeAdditionalFilesFlag,
-        excludeFilesFromAppDataFlag, takeAppOfflineFlag, virtualApplication, setParametersFile, additionalArguments, isParamFilePresentInPackage, isFolderBasedDeployment, 
+        excludeFilesFromAppDataFlag, takeAppOfflineFlag, virtualApplication, setParametersFileName, additionalArguments, isParamFilePresentInPackage, isFolderBasedDeployment, 
         useWebDeploy);
 
+    var errorFile = path.join(tl.getVariable('System.DefaultWorkingDirectory'),"error.txt");
+    var fd = fs.openSync(errorFile, "w");
+    var errObj = fs.createWriteStream("", {fd: fd} );
+     
     try {
-
-        var msDeployBatchFile = tl.getVariable('System.DefaultWorkingDirectory') + '\\' + 'msDeployCommand.bat';
-        var msDeployCommand = '@echo off \n';
-        msDeployCommand += '"' + msDeployPath + '" ' + msDeployCmdArgs + ' 2>error.txt\n';
-        msDeployCommand += 'if %errorlevel% neq 0 exit /b %errorlevel%';
-        tl.writeFile(msDeployBatchFile, msDeployCommand);
-        tl._writeLine(tl.loc("Runningcommand", msDeployCommand));
-        await tl.exec("cmd", ['/C', msDeployBatchFile], <any> {failOnStdErr: true});
-        tl.rmRF(msDeployBatchFile, true);
-        if(publishingProfile != null){
-        tl._writeLine(tl.loc('WebappsuccessfullypublishedatUrl0', publishingProfile.destinationAppUrl));}
+        var msDeployDirectory = msDeployPath.slice(0, msDeployPath.lastIndexOf('\\') + 1);
+        pathVar = process.env.PATH;
+        process.env.PATH = msDeployDirectory + ";" + process.env.PATH ;
```

Comment:
since we ensure that msdeploy path cannot be null, hence we can be sure that msdeploy directory cannot be null